mantis/solver/src/types.rs

```diff
@@ -0,0 +1,203 @@
+//! Basic types with simple checks and domain, no heavy math or solving.
+use crate::prelude::*;
+
+pub type Amount = Decimal;
+
+#[derive(Debug, Clone, Copy, Ord, Eq, PartialEq, PartialOrd, Default)]
+pub struct BuyToken(pub Amount);
+#[derive(Debug, Clone, Copy, Ord, Eq, PartialEq, PartialOrd, Default)]
+pub struct SellToken(pub Amount);
+
+#[derive(Debug, Clone, Copy)]
+pub struct Price(pub Amount);
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum OrderType {
+    BUY,
+    SELL,
+}
+
+impl OrderType {
+    fn as_str(&self) -> &'static str {
+        match self {
+            OrderType::BUY => "Buy",
+            OrderType::SELL => "Sell",
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub enum OrderStatus {
+    PENDING,
+    PARTIALLY_FILLED,
+    FILLED,
```

Comment:
Ditto.
```suggestion
    Pending,
    PartiallyFilled,
    Filled,
```

Reply:
will retain python notation in this case for now